Example/Tests/UserTests.swift

```diff
@@ -177,43 +176,42 @@ class UserTests: QuickSpec {
                 expect(user.tokens).toNot(beNil())
             }
 
-            it("Should emit set event") {
+            it("Should update tokens") {
                 let user = User(state: .loggedOut)
-                var newTokens: TokenData?
-                _ = user.didSetTokens.register { newTokens = $0.new }
+                expect(UserTokensKeychain().data().first).to(beNil())
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe")
-                expect(newTokens).toNot(beNil())
+                expect(UserTokensKeychain().data().first).toNot(beNil())
             }
 
-            it("Should not emit set event if set to same") {
+            it("Should not change tokens if set to same") {
                 let user = User(state: .loggedOut)
-                var newTokens: TokenData?
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe")
-                _ = user.didSetTokens.register { newTokens = $0.new }
+                let tokens = UserTokensKeychain().data().first
+                expect(tokens).toNot(beNil())
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe")
-                expect(newTokens).to(beNil())
+                expect(UserTokensKeychain().data().first).to(equal(tokens))
             }
 
-            it("Should emit set event if any tokens set differently") {
+            it("Should update tokens if set differently") {
                 let user = User(state: .loggedOut)
-                var newTokens: TokenData?
                 _ = try? user.set(accessToken: "hehe", refreshToken: "hehe", idToken: "hehe", userID: "hehe")
-                _ = user.didSetTokens.register { newTokens = $0.new }
-
                 _ = try? user.set(accessToken: "hehe again")
-                expect(newTokens).toNot(beNil())
-                newTokens = nil
+                var tokens = UserTokensKeychain().data().first
+                expect(tokens).toNot(beNil())
+                UserTokensKeychain().removeTokens(forAccessToken: tokens?.accessToken ?? "")
```

Comment:
removeTokens I guess doesn't really do anything here since the tokens are memory based in a user object and only persisted and unpersisted on set/logout right? Should we maybe just use more of user.tokens instead of `UserTokensKeychain().data().first` to get the tokens that are in the user object instead?